dashboard/app/helpers/users_helper.rb

```diff
@@ -21,10 +21,22 @@ def check_and_apply_oauth_takeover(user)
       end
 
       existing_account.destroy! if existing_account
-      user.provider = session['clever_link_flag']
-      user.uid = uid
-      user.oauth_token = session['clever_takeover_token']
-      user.save
+      if user.migrated?
+        user.add_credential(
+          type: session['clever_link_flag'],
+          id: uid,
+          email: user.email,
+          hashed_email: user.hashed_email,
+          data: {
+            oauth_token: session['clever_takeover_token']
+          }
```

Comment:
`AuthenticationOption#data` is still just a text field, and there's no clever before_save behavior to convert a hash to JSON (unless there's some rails magic going on I don't know about).  Do we need to add `.to_json` to the end of this hash?

Reply:
Very good catch.

Worse, this *will* successfully put the record into the database, in a *very* slightly different serialized format that's not quite JSON but looks kinda like it.

I found a couple other cases where we've done this in the code, fixed them up.